### PR TITLE
add azure ipi fips

### PR DESCRIPTION
**Pull request by xiaojiey**

### CODE REVIEW

**Comment by gpei** [comment]
@xiaojiey hi, one quick question on this, as we already have a fips enabled workflow for aws: `aws-ipi-ovn-fips`, do we still need to add one for Azure specifically?

**Comment by xiaojiey** [comment]
> @xiaojiey hi, one quick question on this, as we already have a fips enabled workflow for aws: `aws-ipi-ovn-fips`, do we still need to add one for Azure specifically?

I think enable fips on each platform is mandatory.

**Comment by gpei** [comment]
> > @xiaojiey hi, one quick question on this, as we already have a fips enabled workflow for aws: `aws-ipi-ovn-fips`, do we still need to add one for Azure specifically?
> 
> I think enable fips on each platform is mandatory.
Yeah, I know FIPS could be enabled on every platform, just curious if it's platform independent, which means if a bug for FIPS on Azure also could be reproduced on AWS, then we don't have to test through all the platforms.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fips/provision/cucushift-installer-rehearse-azure-ipi-fips-provision-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+chain:
+  as: cucushift-installer-rehearse-azure-ipi-fips-provision
+  steps:
+  - chain: ipi-conf-azure
+  - chain: ipi-install
```
Comment (gpei) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fips/provision/cucushift-installer-rehearse-azure-ipi-fips-provision-chain.yaml:5]: I noticed there's one existing step [fips-check](https://github.com/openshift/release/tree/master/ci-operator/step-registry/fips-check) defined in the step-registry, do we want to add the ref here

Comment (xiaojiey) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fips/provision/cucushift-installer-rehearse-azure-ipi-fips-provision-chain.yaml:5]: No, the https://github.com/openshift/release/tree/master/ci-operator/step-registry/fips-check comes from one bug(one related to real node is fips disabled on rhel nodes). The step will check whether fips enabled for rhel nodes before install ocp packages, if not, will refuse to do more installation.

Comment (gpei) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fips/provision/cucushift-installer-rehearse-azure-ipi-fips-provision-chain.yaml:5]: ok, thanks for the info

**Comment by xiaojiey** [comment]
> @xiaojiey hi, one quick question on this, as we already have a fips enabled workflow for aws: `aws-ipi-ovn-fips`, do we still need to add one for Azure specifically?

I think enable fips on each platform is mandatory. 

> @xiaojiey hi, one quick question on this, as we already have a fips enabled workflow for aws: `aws-ipi-ovn-fips`, do we still need to add one for Azure specifically?

We have this one https://issuetracker.google.com/issues/197769045 related to rhel nodes. But I remembered storage team has some test cases could not run on azure with fips enabled cluster as algorithm not compliant to fips.

**Pull request merged**